### PR TITLE
Sle 15 sp4

### PR DESCRIPTION
**Pull request by varkoly**
## Problem

*The local variable secondary_host_name was not set which was required in the adapted template.*
*The primary and secondary hostname was badly detected*
*Furthermore there was a syntax error in the template: missing ":" after mandatory*

## Solution

Set the local variable  secondary_host_name.
Create new function get_primary_on_primary to find the primary hostname.
Fix syntax error in template.

### CODE REVIEW

File: src/lib/sap_ha/configuration/cluster.rb [hunk 1]
```diff
@@ -278,6 +278,16 @@ def other_nodes_ext
         end.compact
       end
 
+      def get_primary_on_primary
+        SapHA::System::Network.ip_addresses.each do |my_ip|
+	  @nodes.map do |_, node|
```
Comment (lslezak) [inline src/lib/sap_ha/configuration/cluster.rb:283]: This `map` call is useless, the result is not (or should not as it seems) used as the result of this function. You can use `each` here.

File: src/lib/sap_ha/configuration/cluster.rb [hunk 2]
```diff
@@ -278,6 +278,16 @@ def other_nodes_ext
...
+	    if node[:ip_ring1] == my_ip
+	      return node[:host_name]
+	    end
+	  end
+	end
```
Comment (lslezak) [inline src/lib/sap_ha/configuration/cluster.rb:288]: Be careful, if that `return` above is not called then the result of the function is the result of the top level `each` call. And that is the `SapHA::System::Network.ip_addresses` list itself.

To be safe I'd add `return nil` here. Or if that case should not really happen you might raise an exception here which will be caught by YaST top level exception handler and an error will be reported to user.

**Review by lslezak — COMMENTED** [review]
Small suggestions

File: src/lib/sap_ha/configuration/cluster.rb (hunk 2 above)
Comment (lslezak) [reply src/lib/sap_ha/configuration/cluster.rb:288]: BTW here is short summary about `return`, `break` and `next` in blocks: https://makandracards.com/makandra/46939-ruby-a-small-summary-of-what-return-break-and-next-means-for-blocks

Comment (varkoly) [reply src/lib/sap_ha/configuration/cluster.rb:288]: Thanks. Proposed change applied.

File: src/lib/sap_ha/configuration/cluster.rb (hunk 1 above)
Comment (varkoly) [reply src/lib/sap_ha/configuration/cluster.rb]: Thanks

**Review by lslezak — APPROVED** [review]
Thank you!

**Pull request merged**